src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectLockFileWatcher.cs

```diff
@@ -159,17 +159,27 @@ public int FilesChanged(uint cChanges, string[] rgpszFile, uint[] rggrfChange)
             // the kind of change since we are interested in all changes.
             _projectServices.ThreadingService.Fork(async () =>
             {
-                using (var access = await _projectLockService.WriteLockAsync())
+                try
                 {
-                    // notify all the loaded configured projects
-                    var currentProjects = _projectServices.Project.LoadedConfiguredProjects;
-                    foreach (var configuredProject in currentProjects)
+                    await _projectServices.Project.Services.ProjectAsynchronousTasks.LoadedProjectAsync(async () =>
                     {
-                        // Inside a write lock, we should get back to the same thread.
-                        var project = await access.GetProjectAsync(configuredProject).ConfigureAwait(true);
-                        project.MarkDirty();
-                        configuredProject.NotifyProjectChange();
-                    }
+                        using (var access = await _projectLockService.WriteLockAsync())
+                        {
+                            // notify all the loaded configured projects
+                            var currentProjects = _projectServices.Project.LoadedConfiguredProjects;
+                            foreach (var configuredProject in currentProjects)
+                            {
+                                // Inside a write lock, we should get back to the same thread.
```

Comment:
This comment isn't strictly true. There's no guarantee about the thread you return after an await being the same one you started on. But as you're on a threadpool thread due to the project lock, you will return to the threadpool (but any thread within it).